File: articles/decision-science-judgment-under-uncertainty/go/expected_value.go
package main

import "fmt"

type Outcome struct {
	Probability float64
	Value       float64
}

func expectedValue(outcomes []Outcome) float64 {
	total := 0.0

	for _, outcome := range outcomes {
		total += outcome.Probability * outcome.Value
	}

	return total
}

func main() {
	outcomes := []Outcome{
		{Probability: 0.65, Value: 72.0},
		{Probability: 0.35, Value: 38.0},
	}

	fmt.Println("Expected value:")
	fmt.Printf("%.3f\n", expectedValue(outcomes))
}

File: articles/decision-science-judgment-under-uncertainty/c/expected_value.c
#include <stdio.h>

int main(void) {
    double probabilities[] = {0.65, 0.35};
    double values[] = {72.0, 38.0};
    int n = 2;
    double expected_value = 0.0;

    for (int i = 0; i < n; i++) {
        expected_value += probabilities[i] * values[i];
    }

    printf("Expected value: %.3f\n", expected_value);

    return 0;
}
